thirdeye-server/src/main/java/ai/startree/thirdeye/resources/CrudResource.java

```diff
@@ -110,6 +113,11 @@ private DtoT updateDto(final ThirdEyePrincipal principal, final ApiT api) {
 
     final DtoT updated = toDto(api);
 
+    // Check the that user can modify the resource before and after the update.
```

Comment:
Longer term we need to refactor this setup entirely I feel. but thanks for writing this in such a concise fashion. 👍